adapter/src/main/java/com/twilio/kudu/sql/mutation/CubeMaintainer.java

```diff
@@ -91,12 +93,12 @@ public long getFloorMod() {
    * to the cube rollup time
    * 
    * @param colIndexToValueMap map from fact table column index to column value
-   * @return pair of ( map from cube table pk column index to column value, map
-   *         from cube table non-pk column index to column value)
+   * @return pair of ( map from cube table pk column index to column value, array
+   *         of non-pk column values)
    */
-  public Pair<Map<Integer, Object>, Map<Integer, Object>> generateCubeDelta(Map<Integer, Object> colIndexToValueMap) {
```

Comment:
Save space by deriving the indexes of the non pk columns from `nonPkColIndexMap`  instead of storing the indexes per row.